### PR TITLE
Alphafold datacheck

### PR DESCRIPTION
**Pull request by alisha17**

### CODE REVIEW

**Review by marcoooo — CHANGES_REQUESTED** [review]
1. please reopen against 107, since we are supposed to use it for current started release.
2. I don't believe the comparison is all right, I ran those query against human for 106 and the counts don't match. 

```select count(*) from protein_feature pf, analysis a where a.analysis_id = pf.analysis_id and a.logic_name = 'alphafold_import';``` 
=> 32813

```select count(*) from protein_feature pf, analysis a where a.analysis_id = pf.analysis_id and a.logic_name = 'alphafold_import' and pf.hit_name like 'AF-%-F1._'```
=> 29923

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckAlphaFoldFormat',
+  DESCRIPTION    => 'Check records for alphafold import',
+  GROUPS         => ['protein_feature'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['core'],
+  TABLES         => ['protein_feature'],
```
Comment (marcoooo) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm:36]: Analysis is mentioned as well in the query, you may add `analysis` in the list then.

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm (hunk 1 above)
Comment (alisha17) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm]: Ok added

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  is_rows_nonzero($self->dba, $sql_1, $desc_1);
+
+
+  my $desc_2 = "All Alpha fold records with specific format";
+  my $sql_2  = q/
+    select count(*) from protein_feature pf, analysis a where a.analysis_id = pf.analysis_id and a.logic_name = 'alphafold_import' and pf.hit_name  REGEXP 'AF\-[A-Za-z0-9]+\-F[0-9]+\.[A-Z]'
+  /;
+  is_rows($self->dba, $sql_2, 1, $desc_2);
+
+  my $des = "compare count of alpha fold records with all alpha fold records of specific format"
```
Comment (marcoooo) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm:55]:
```suggestion
  my $des = "All alpha fold analysis records share the right format"
```

Basically messages should be a confirmation of success test.

**Review by marcoooo — CHANGES_REQUESTED** [review]

**Comment by marcoooo** [comment]
@alisha17 now you hav eto fixthe build: 
```t/index.t ............ syntax error at /home/travis/build/Ensembl/ensembl-datacheck/lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm line 56, near "my "

Global symbol "$dbc" requires explicit package name at /home/travis/build/Ensembl/ensembl-datacheck/lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm line 56.

Global symbol "$dbc" requires explicit package name at /home/travis/build/Ensembl/ensembl-datacheck/lib/Bio/EnsEMBL/DataCheck/Checks/CheckAlphafoldEntries.pm line 57.

Compilation failed in require at /home/travis/build/Ensembl/ensembl-datacheck/lib/Bio/EnsEMBL/DataCheck/Manager.pm line 356.

Dubious, test returned 255 (wstat 65280, 0xff00)```

**Review by marcoooo — APPROVED** [review]

**Pull request merged**